File: Toggly.FeatureManagement.Nuxt/nuxt-toggly-core/tests/smoke.test.ts
import { describe, it, expect } from 'vitest';
import { createTogglyClient } from '../src/client';

describe('Smoke test', () => {
  const appKey = process.env.TOGGLY_SMOKE_APP_KEY_FRONTEND;

  it('loads live evaluated flags', async () => {
    if (!appKey) {
      return;
    }

    const client = createTogglyClient({
      appKey,
      environment: 'Production',
      baseUri: 'https://definitions.toggly.io',
      refreshInterval: 0,
    });

    await client.init();

    await expect(client.isFeatureOn('FlagOn')).resolves.toBe(true);
    await expect(client.isFeatureOff('FlagOff')).resolves.toBe(true);
  });
});

File: Toggly.FeatureManagement.Android/toggly-android-core/src/test/kotlin/io/toggly/core/SmokeTest.kt
package io.toggly.core

import io.toggly.core.models.TogglyConfig
import kotlinx.coroutines.delay
import kotlinx.coroutines.test.runTest
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test
import kotlin.time.Duration.Companion.seconds

class SmokeTest {

    @Test
    fun `loads live evaluated flags`() = runTest(timeout = 20.seconds) {
        val appKey = System.getenv("TOGGLY_SMOKE_APP_KEY_FRONTEND")
        if (appKey.isNullOrBlank()) {
            return@runTest
        }

        val service = TogglyService(
            TogglyConfig(
                appKey = appKey,
                environment = "Production",
                baseUri = "https://definitions.toggly.io",
                refreshInterval = 0
            )
        )

        service.init()

        assertTrue(service.isFeatureOn("FlagOn"))
        assertTrue(service.isFeatureOff("FlagOff"))
        assertFalse(service.isFeatureOn("FlagOff"))
    }

    @Test
    fun `WebSocket connects and flags are correct`() = runTest(timeout = 20.seconds) {
        val appKey = System.getenv("TOGGLY_SMOKE_APP_KEY_FRONTEND")
        if (appKey.isNullOrBlank()) {
            return@runTest
        }

        val service = TogglyService(
            TogglyConfig(
                appKey = appKey,
                environment = "Production",
                baseUri = "https://definitions.toggly.io",
                refreshInterval = 0,
                enableLiveUpdates = true
            )
        )

        service.init()

        // Brief wait for WebSocket to connect
        delay(2_000)

        assertTrue(service.isFeatureOn("FlagOn"))
        assertTrue(service.isFeatureOff("FlagOff"))
        assertFalse(service.isFeatureOn("FlagOff"))
    }
}
